bytes/bytes/api/router.py

```diff
@@ -148,42 +148,59 @@ def get_normalizer_meta(
 
 @router.post("/raw", tags=[RAW_TAG])
 def create_raw(
-    request: Request,
     boefje_meta_id: UUID,
-    mime_types: list[str] | None = Query(None),
+    boefje_output: BoefjeOutput,
     meta_repository: MetaDataRepository = Depends(create_meta_data_repository),
     event_manager: EventManager = Depends(create_event_manager),
-) -> RawResponse:
-    parsed_mime_types = [] if mime_types is None else [MimeType(value=mime_type) for mime_type in mime_types]
-
-    try:
-        meta = meta_repository.get_boefje_meta_by_id(boefje_meta_id)
-
-        if meta_repository.has_raw(meta, parsed_mime_types):
-            return RawResponse(status="success", message="Raw data already present")
-
-        # FastAPI/starlette only has async versions of the Request methods, but
-        # all our code is sync, so we wrap it in async_to_sync.
-        data = async_to_sync(request.body)()
-
-        raw_data = RawData(value=data, boefje_meta=meta, mime_types=parsed_mime_types)
-        with meta_repository:
-            raw_id = meta_repository.save_raw(raw_data)
-
-        event = RawFileReceived(
-            organization=meta.organization,
-            raw_data=RawDataMeta(
-                id=raw_id,
-                boefje_meta=raw_data.boefje_meta,
-                mime_types=raw_data.mime_types,
-            ),
-        )
-        event_manager.publish(event)
-    except Exception as error:
-        logger.exception("Error saving raw data")
-        raise HTTPException(status_code=500, detail="Could not save raw data") from error
-
-    return RawResponse(status="success", message="Raw data saved", id=raw_id)
+) -> JSONResponse:
+    """Parse all the raw files from the request and return the ids. The ids are ordered according to the order from the
+    request data, but we assume the `name` field is unique, and hence return a mapping of the file name to the id."""
+
+    raw_ids = {}
+    mime_types_by_id = {
+        raw.id: set(raw.mime_types) for raw in meta_repository.get_raw(RawDataFilter(boefje_meta_id=boefje_meta_id))
+    }
+    all_parsed_mime_types = list(mime_types_by_id.values())
+
+    for raw in boefje_output.files:
+        parsed_mime_types = {MimeType(value=x) for x in raw.tags}
+
+        if parsed_mime_types in mime_types_by_id.values():
+            # Set the id for this file using the precomputed dict that maps existing primary keys to the mime-type set.
+            raw_ids[raw.name] = str(
+                list(mime_types_by_id.keys())[list(mime_types_by_id.values()).index(parsed_mime_types)]
+            )
+            continue
+
+        if parsed_mime_types in all_parsed_mime_types:
+            raise HTTPException(status_code=400, detail="Content types do not define unique sets of mime types.")
```

Comment:
Please use the status codes constants defined in `httpx.codes`

```suggestion
            raise HTTPException(status_code=codes.NOT_FOUND, detail="Content types do not define unique sets of mime types.")
```